lib/public/components/common/form/inputs/DurationInputModel.js

```diff
@@ -0,0 +1,136 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+
+import { Observable } from '/js/src/index.js';
+
+/**
+ * @typedef DurationInputRawData
+ * @property {number} hours the number of hours
+ * @property {number} minutes the number of minutes
+ * @property {number} seconds the number of seconds
+ */
+
+/**
+ * Store the duration input
+ */
+export class DurationInputModel extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+
+        this._raw = {
+            hours: null,
+            minutes: null,
+            seconds: null,
+        };
+
+        /**
+         * Timestamp (ms)
+         * @type {number|null}
+         * @private
+         */
+        this._value = null;
+    }
+
+    /**
+     * Update the inputs raw values
+     * @param {DurationInputRawData} raw the input raw values
+     * @return {void}
+     */
+    update(raw) {
+        try {
+            const previousVal = this._raw;
+            this._raw = { ...this._raw, ...raw };
+
+            if (this._raw.seconds === 60) {
+                this._raw.minutes++;
+                this._raw.seconds = 0;
+            }
+            if (this._raw.minutes === 60) {
+                this._raw.hours++;
+                this._raw.minutes = 0;
+            }
+
+            if (this._raw.seconds === -1) {
+                this._raw.minutes--;
+                this._raw.seconds = 59;
+            }
+            if (this._raw.minutes === -1) {
+                this._raw.hours--;
+                this._raw.minutes = 59;
+            }
+
+            if (this._raw.hours < 0) {
+                this.update(previousVal);
+                return;
+            }
+            this._value =
+                (this._raw.hours || 0) * 60 * 60 * 1000 +
+                (this._raw.minutes || 0) * 60 * 1000 +
+                (this._raw.seconds || 0) * 1000;
+
+            if (this._value === 0) {
```

Comment:
Why?

Reply:
To me it's not more elegant, `0` is not `null`. You may want to look for runs that were stopped as soon as they were created, or you may use duration for other things that may have a 0 duration?